### ISSUE TITLE
Panic on Method that Without Block to Yield

### ISSUE THREAD

**Issue by Maxwell-Alexius** [goby-lang/goby#273]
While I'm trying to add test case for some methods like `Hash#each_key`, `Hash#each_value`, `Hash#map_values`...etc which requires a block to yield. There is a case when developer don't give a block, it does print out the error message:

```
Internal Error: Can't yield without a block
```

However, it still panic the program and exited from REPL.

```
$ go run goby.go -i
Goby 0.0.9
» [1, 2, 3].each do |v|
*   puts v
» end
1
2
3
#=> [1, 2, 3]
» [1, 2, 3].each
Internal Error: Can't yield without a block
panic: (vm.errorMessage) (0x1320000,0xc420150930) [recovered]
        panic: (vm.errorMessage) (0x1320000,0xc420150930)

goroutine 1 [running]:
...... Error messages being omitted .......
exit status 2
```

**Comment by st0012** [comment]
@Maxwell-Alexius I modified the behavior, can you rebase and try again?

**Comment by Maxwell-Alexius** [comment]
I'm not sure, the error message pop out like this:

```
$ go run goby.go -i
Goby 0.0.9
» [1, 2, 3].each do |v|
*   puts v
» end
1
2
3
#=> [1, 2, 3]
» [1, 2, 3].each
Internal Error: runtime error: invalid memory address or nil pointer dereference
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12a0af6]

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*thread).execInstruction.func1(0xc420077d40)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:71 +0x81
panic(0x133b380, 0x1540260)
        /usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
github.com/goby-lang/goby/vm.(*thread).builtInMethodYield(0xc420077d40, 0x0, 0xc42004b6b0, 0x1, 0x1, 0x100dd01)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:79 +0x26
github.com/goby-lang/goby/vm.builtinArrayInstanceMethods.func7.1(0xc420077d40, 0x1566a78, 0x0, 0x0, 0x0, 0xc42011c1e0, 0xc4201502ca)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/array.go:321 +0xcc
github.com/goby-lang/goby/vm.(*thread).evalBuiltInMethod(0xc420077d40, 0x151e5e0, 0xc4201503c0, 0xc42011e560, 0x0, 0x0, 0x1, 0x0)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:130 +0x202
github.com/goby-lang/goby/vm.glob..func21(0xc420077d40, 0xc4201aa5b0, 0xc420150380, 0x2, 0x2)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/instruction.go:363 +0x397
github.com/goby-lang/goby/vm.(*thread).execInstruction(0xc420077d40, 0xc4201aa5b0, 0xc420130de0)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:75 +0x96
github.com/goby-lang/goby/vm.(*thread).evalCallFrame(0xc420077d40, 0xc4201aa5b0)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:42 +0x60
github.com/goby-lang/goby/vm.(*thread).startFromTopFrame(0xc420077d40)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/thread.go:36 +0x55
github.com/goby-lang/goby/vm.(*VM).startFromTopFrame(0xc420102120)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/vm.go:192 +0x2f
github.com/goby-lang/goby/vm.(*VM).REPLExec(0xc420102120, 0xc4201b0008, 0x1, 0x1)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/vm/repl.go:43 +0x66a
github.com/goby-lang/goby/igb.StartIgb(0x139f73f, 0x5)
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/igb/repl.go:204 +0x1054
main.main()
        /Users/MaxwellAlexius/go/src/github.com/goby-lang/goby/goby.go:27 +0x657
exit status 2
```

In PR #267, I think another solution is to use this (which I also commented on that PR also):

```go
initErrorObject(InternalErrorClass, "Can't yield without a block")
```

What do you suggest?

**Comment by st0012** [comment]
@Maxwell-Alexius The second panic is not raised by Goby, it's because there's a bug in program so you should follow the trace and see what causes this.

**Comment by Maxwell-Alexius** [comment]
Okay, I'll check it out > <

**Comment by st0012** [comment]
This should be fixed in #284

**Comment by Maxwell-Alexius** [comment]
Closed with confirmed fixed in PR #284 which is also being merged.
